homeassistant/components/color_extractor/__init__.py

```diff
@@ -88,12 +100,19 @@ async def async_handle_service(service_call):
             )
             return
 
-        if color:
-            service_data[ATTR_RGB_COLOR] = color
+        if colors:
+            if isinstance(service_data[ATTR_ENTITY_ID], list):
+                lights = service_data[ATTR_ENTITY_ID]
+            else:
+                lights = [service_data[ATTR_ENTITY_ID]]
 
-            await hass.services.async_call(
-                LIGHT_DOMAIN, LIGHT_SERVICE_TURN_ON, service_data, blocking=True
-            )
+            for _, (entity_id, color) in enumerate(zip(lights, colors)):
+                service_data[ATTR_ENTITY_ID] = entity_id
+                service_data[ATTR_RGB_COLOR] = color
```

Comment:
You need to create a copy. You don't know how long this dict will stay around. 
```suggestion
                data = {
                  **service_data,
                  ATTR_ENTITY_ID: entity_id,
                  ATTR_RGB_COLOR: color,
                }
```